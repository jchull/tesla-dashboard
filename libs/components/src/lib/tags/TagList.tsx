import React from 'react'
import './tag.style.scss'
import { Tag } from './Tag'
import { TagInput } from './TagInput'

interface TagListProps {
  tags: string[]

  addListener(tag: string)

  removeListener(tag: string)
}

export const TagList: React.FC<TagListProps> = (props: TagListProps) => {
  const tags = props.tags.map((tag) => (
    <Tag value={tag}
         removeListener={props.removeListener}
         key={tag} />
  ))
  return (
    <div className='card tag-list'>
      <h5>Tags</h5>
      {tags}
      {props.addListener && <TagInput addListener={props.addListener} />}
    </div>
  )
}
