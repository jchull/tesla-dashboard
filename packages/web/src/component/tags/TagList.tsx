import React from 'react';
import './tag.style.scss';
import { Tag } from './Tag';
import { TagInput } from './TagInput';

interface TagListProps {
  tags: string[];

  addListener(tag: string): void;

  removeListener(tag: string): void;
}

export const TagList: React.FC<TagListProps> = (props: TagListProps) => {
  const tags = props.tags.map((tag) => (
    <Tag value={tag} removeListener={props.removeListener} key={tag} />
  ));
  return (
    <div className="card tag-list">
      <h5>Tags</h5>
      {tags}
      {props.addListener && <TagInput addListener={props.addListener} />}
    </div>
  );
};
