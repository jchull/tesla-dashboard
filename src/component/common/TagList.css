
.tag {
    display: inline-block;
    height: 40px;
    line-height: 36px;
    border-radius: 40px;
    margin: 2px;
    padding: 0 6px 0 12px;
    vertical-align: top;
}

.tag span {
    vertical-align: top;
    font-size: 20px;
}

.tag-list button {
    width: 30px;
    height: 30px;
    border-radius: 30px;
    padding: 0;
    background: #f7f7f7;
    margin: 3px 0 3px 10px;
    font-size: 24px;
}

.tag-list button.add {
    margin-top: 6px;
}

.tag-list button:hover {
    color: #2a9de6;
}


.tag-list .tag button.remove:hover {
    border: 1px solid rgb(227, 25, 55);
    color: rgb(227, 25, 55);
}

.tag-list button:focus {
    outline: none;
}

.tag-list button i {
    font-size: 24px;
    line-height: 29px;
    margin: 0;
    background: none;
}


.tag-input {
    display: inline-block;
}

.tag-input input:focus {
    outline: none;
}

.tag-input input {
    border-radius: 20px;
    line-height: 24px;
    vertical-align: bottom;
    font-size: 20px;
    padding: 4px 10px;
    margin: 0 4px;
}

.tag-input input.hide {
    transition-duration: 250ms;
    width: 0;
    border-color: transparent;
}

.tag-input input.show {
    transition-duration: 250ms;
    width: 200px;
}
