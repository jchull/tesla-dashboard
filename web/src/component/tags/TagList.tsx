import React from 'react';
import './tag.style.scss';
import {Tag} from './Tag';
import {TagInput} from './TagInput';

interface TagListProps {
  tags: string[];

  addListener(tag: string): void;

  removeListener(tag: string): void;
}

export const TagList: React.FC<TagListProps> = (props: TagListProps) => {

  const tags = props.tags.map(tag => <Tag value={tag}
                                          removeListener={props.removeListener}
                                          key={tag}/>);
  return (
      <div className="tag-list">
        <div>Tags</div>
        {tags}
        {props.addListener &&
        <TagInput addListener={props.addListener}/>
        }
      </div>
  );
};

