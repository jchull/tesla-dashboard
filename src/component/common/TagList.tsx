import React from 'react';
import './TagList.css';
import {Tag} from './Tag';
import {TagInput} from './TagInput';

interface TagListProps {
  tags: Array<string>;

  addListener(tag: string): any;

  removeListener(tag: string): any;
}

export const TagList: React.FC<TagListProps> = (props: TagListProps) => {

  const tags = props.tags.map(tag => <Tag value={tag}
                                          removeListener={props.removeListener}
                                          key={tag}/>);
  return (
      <div className="tag-list">
        {tags}
        {props.addListener &&
          <TagInput addListener={props.addListener}/>
        }
      </div>
  );
};

