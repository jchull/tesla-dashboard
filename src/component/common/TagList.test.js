import React                              from 'react';
import { render, unmountComponentAtNode } from 'react-dom';
import { act } from "react-dom/test-utils";

import { TagList } from './TagList';

let container = null;
beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});


it('renders empty tag list', () => {
  act(() => {
    render(<TagList tags={[]}
                    addListener={console.log}
                    removeListener={console.log}/>, container);
  });
  // expect(container.innerHTML)
  //   .toMatchInlineSnapshot();
});

it('renders simple tag list', () => {
  act(() => {
    render(<TagList tags={['item_1', 'item_2']}
                    addListener={console.log}
                    removeListener={console.log}/>, container);
  });
  // expect(container.innerHTML)
  //   .toMatchInlineSnapshot();
});
