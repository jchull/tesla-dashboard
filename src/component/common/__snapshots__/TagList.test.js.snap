// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders empty tag list 1`] = `
"<div class=\\"tag-list\\">
  <div class=\\"tag-input\\"><button class=\\"add\\" aria-label=\\"Show new tag entry\\"><i class=\\"material-icons\\">add</i></button><input class=\\"hide\\" aria-label=\\"Enter new tag\\"></div>
</div>"
`;

exports[`renders simple tag list 1`] = `
"<div class=\\"tag-list\\">
  <div class=\\"tag\\" style=\\"border: 2px solid #f4f4f4;\\"><span>item_1</span><button class=\\"remove\\"><i class=\\"material-icons\\">remove</i></button></div>
  <div class=\\"tag\\" style=\\"border: 2px solid #f4f4f4;\\"><span>item_2</span><button class=\\"remove\\"><i class=\\"material-icons\\">remove</i></button></div>
  <div class=\\"tag-input\\"><button class=\\"add\\" aria-label=\\"Show new tag entry\\"><i class=\\"material-icons\\">add</i></button><input class=\\"hide\\" aria-label=\\"Enter new tag\\"></div>
</div>"
`;
